src/rational_bezier_spline.hpp

```diff
@@ -262,27 +262,45 @@ class RationalBezierSpline {
     return Evaluate(PointTypeParametric_{par_coords...});
   }
 
-  /// Evaluate Basis Functions
+  /// Evaluate Basis Functions in the form (w_i N_i / (sum(w_j N_j)))
   template <typename... T>
-  constexpr std::array<std::vector<ScalarType>, parametric_dimension>
-  BasisFunctions(const T&... par_coords) const {
+  constexpr std::vector<ScalarType> BasisFunctions(
+      const T&... par_coords) const {
     return BasisFunctions(PointTypeParametric_{par_coords...});
   }
 
   /// Evaluate Basis Functions
-  constexpr std::array<std::vector<ScalarType>, parametric_dimension>
-  BasisFunctions(const PointTypeParametric_& par_coords) const;
+  constexpr std::vector<ScalarType> BasisFunctions(
+      const PointTypeParametric_& par_coords) const;
+
+  /**
+   * @brief Evaluate non weighted basis functions
+   *
+   * returns basis function values in the form (N_i / (sum(w_j N_j)))
+   * They are required in the use of weighted control points to avoid
+   * multiplication followed by division of the same coefficients. For internal
+   * use only
+   */
+  template <typename... T>
+  constexpr std::vector<ScalarType> NonWeightedBasisFunctions(
+      const T&... par_coords) const {
+    return NonWeightedBasisFunctions(PointTypeParametric_{par_coords...});
+  }
+
+  /// Evaluate Basis Functions
+  constexpr std::vector<ScalarType> NonWeightedBasisFunctions(
+      const PointTypeParametric_& par_coords) const;
 
   /// Evaluate Basis Functions
   template <typename... T>
   constexpr std::array<std::vector<ScalarType>, parametric_dimension>
-  PolynomialBasisFunctions(const T&... par_coords) const {
-    return PolynomialBasisFunctions(PointTypeParametric_{par_coords...});
+  BasisFunctionContributions(const T&... par_coords) const {
```

Comment:
Contribution means, basis values per dimension?

Reply:
yes, it is just

$$ R_{ijk}(\xi, \eta, \zeta) = B_{i}(\xi)B_{j}(\eta)B_{k}(\zeta)$$